testing/scenario_app/test/adb_log_filter_test.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:litetest/litetest.dart';
+
+import '../bin/utils/adb_logcat_filtering.dart';
+import 'src/fake_adb_logcat.dart';
+
+void main() {
+  /// Simulates the filtering of logcat output [lines].
+  Iterable<String> filter(Iterable<String> lines, {int? filterProcessId}) {
```

Comment:
nit: does this really belong in its own testable class? Right now this looks like it's roughly copied from the run_android_test.dart

Reply:
I'll file a follow-up issue, as it really depends on how much more work I'm going to do on the CLI:
https://github.com/flutter/flutter/issues/144252